lib/blueprinter-activerecord/preloader.rb

```diff
@@ -62,22 +63,29 @@ def pre_render(object, blueprint, view, options)
     #
     # Example:
     #
-    #   preloads = BlueprinterActiveRecord::Preloader.preloads(WidgetBlueprint, :extended, Widget)
+    #   preloads = BlueprinterActiveRecord::Preloader.preloads(WidgetBlueprint, :extended, model: Widget)
     #   q = Widget.where(...).order(...).preload(preloads)
     #
     # @param blueprint [Class] The Blueprint class
     # @param view_name [Symbol] Name of the view in blueprint
     # @param model [Class] The ActiveRecord model class that blueprint represents
     # @return [Hash] A Hash containing preload/eager_load/etc info for ActiveRecord
     #
-    def self.preloads(blueprint, view_name, model=nil)
+    def self.preloads(blueprint, view_name, options = {}, model: nil, recursion_level: 0)
       view = blueprint.reflections.fetch(view_name)
       preload_vals = view.associations.each_with_object({}) { |(_name, assoc), acc|
         # look for a matching association on the model
         ref = model ? model.reflections[assoc.name.to_s] : nil
         if (ref || model.nil?) && !assoc.blueprint.is_a?(Proc)
-          ref_model = ref && !(ref.belongs_to? && ref.polymorphic?) ? ref.klass : nil
-          acc[assoc.name] = preloads(assoc.blueprint, assoc.view, ref_model)
+          recursive = assoc.blueprint == blueprint && assoc.view == view_name
```

Comment:
Is there a possibility of us running into this issue when the views _aren't_ the same?

Reply:
Probably. After some more testing, I think we need a more general cycle detection algorithm, e.g. BlueprintA/:default -> BlueprintB/:default -> BlueprintA/:default -> BlueprintB/:default -> ...

Going to try to work that in here.